pytorch_lightning/strategies/bagua.py

```diff
@@ -0,0 +1,239 @@
+        log.info(
+            "Initializing Bagua Distributed: "
+            f"GLOBAL_RANK: {self.global_rank}, "
+            f"MEMBER: {self.global_rank + 1}/{self.world_size}"
+        )
+
+        # need to set device first before initialize Bagua distributed environment
+        torch.cuda.set_device(self.local_rank)
+
+        if not is_initialized():
+            bagua.init_process_group()
+
+    def _set_node_environment_variables(self) -> None:
+        os.environ["MASTER_ADDR"] = self.cluster_environment.main_address  # type: ignore[union-attr]
+        os.environ["MASTER_PORT"] = str(self.cluster_environment.main_port)  # type: ignore[union-attr]
+        os.environ["RANK"] = str(self.global_rank)
+        os.environ["NODE_RANK"] = str(self.node_rank)
+        os.environ["WORLD_SIZE"] = str(self.world_size)
+        os.environ["LOCAL_RANK"] = str(self.local_rank)
+
+    def _check_qadam_optimizer(self) -> None:
+
+        has_qadam_optimizer = any([isinstance(opt, QAdamOptimizer) for opt in self.optimizers])
+
+        if not has_qadam_optimizer or len(self.optimizers) > 1 or len(self.lr_schedulers) > 1:
+            raise MisconfigurationException("Bagua QAdam can only accept one QAdamOptimizer and one LR Scheduler.")
+
+    def configure_ddp(self) -> None:
+        model = LightningBaguaModule(self.model)  # type: ignore[arg-type]
+        self._model = self._setup_model(model)
+
+        # start the background communication for async algorithm
+        if self.lightning_module.trainer.training and self._bagua_algorithm == "async":
+            self.model.bagua_algorithm.resume(self.model)  # type: ignore
+
+    def _setup_model(self, model: Module) -> BaguaDistributedDataParallel:
+        """Wraps the model into a Bagua distributed module."""
+
+        if self._bagua_algorithm == "qadam":
+            self._check_qadam_optimizer()
+            self._bagua_kwargs["q_adam_optimizer"] = self.optimizers[0]
+
+        algorithm = Algorithm.init(self._bagua_algorithm, **self._bagua_kwargs)
+        return BaguaDistributedDataParallel(
+            module=model,
+            optimizers=self.optimizers,
+            algorithm=algorithm,
+            gradient_as_bucket_view=self._bagua_do_flatten,
+        )
+
+    @property
+    def lightning_module(self) -> "pl.LightningModule":
+        model = self._model
+        if isinstance(model, BaguaDistributedDataParallel):
+            model = model.module
+        return unwrap_lightning_module(model)  # type: ignore[arg-type]
+
+    @classmethod
+    def register_plugins(cls, plugin_registry: Dict) -> None:
+        plugin_registry.register("bagua", cls, description="Default Bagua Plugin")
+
+    def teardown(self) -> None:
+        # abort the background communication for async algorithm
+        if self.lightning_module.trainer.training and self._bagua_algorithm == "async":
+            self.model.bagua_algorithm.abort(self.model)  # type: ignore
+
```

Comment:
Shouldn't this also do the unwrapping of `DistributedDataParallel` as in the parent?